actionview/lib/action_view/helpers/tags/label.rb

```diff
@@ -58,8 +58,8 @@ def render(&block)
 
           add_default_name_and_id_for_value(tag_value, name_and_id)
           options.delete("index")
-          options.delete("namespace")
-          options["for"] = name_and_id["id"] unless options.key?("for")
+          namespace = options.delete("namespace")
+          options["for"] = name_and_id["id"] if (namespace.present? && options.key?("for")) || !options.key?("for")
```

Comment:
Should this just be

```suggestion
          options["for"] = name_and_id["id"] if namespace || !options.key?("for")
```

since the namespace option means we need the "namespace prefixed" value from `add_default_and_id_for_value`

Reply:
I like this suggestion as it simplifies the fix. I checked it locally and all actionview tests still pass